src/api/app/controllers/announcements_controller.rb

```diff
@@ -1,73 +1,64 @@
 class AnnouncementsController < ApplicationController
-  before_action :set_announcement, only: [:show, :update, :destroy]
+  before_action :set_status_message, only: [:show, :update, :destroy]
```

Comment:
Are you missing `require_admin`?

Reply:
IMO, this is covered Pundit (status_message_policies). unless I miss something.